src/NServiceBus.Core.Tests/Recoverability/DefaultRecoverabilityPolicyTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace NServiceBus.Core.Tests.Recoverability
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Transport;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class DefaultRecoverabilityPolicyTests
+    {
+        [Test]
+        public void When_max_immediate_reties_have_not__been_reached_should_return_immediate_retry_action()
+        {
+            var policy = CreatePolicy(maxImmediateRetries: 3);
+            var errorContext = CreateErrorContext(numberOfDeliveryAttempts: 2);
+
+            var recoverabilityAction = policy.Invoke(errorContext);
+
+            Assert.IsInstanceOf<ImmediateRetry>(recoverabilityAction, "We should have one immediate retry left. It is second delivery attempt and we configured immediate reties to 2.");
+        }
+
+        [Test]
+        public void When_max_immediate_retries_exceeded_should_return_action_other_than_immediate_retry()
+        {
+            var policy = CreatePolicy(maxImmediateRetries: 1);
+            var errorContext = CreateErrorContext(numberOfDeliveryAttempts: 3);
+
+            var recoverabilityAction = policy.Invoke(errorContext);
+
+            Assert.IsNotInstanceOf<ImmediateRetry>(recoverabilityAction, "When max number of immediate retries exceeded should return action other than ImmediateRetry.");
+        }
+
+        [Test]
+        public void When_immediate_retries_turned_off_and_slr_policy_returns_delay_should_return_delayed_retry_action()
+        {
+            var deliveryDelay = TimeSpan.FromSeconds(10);
+            var policy = CreatePolicy(maxImmediateRetries: 0, delayedRetryDelay: deliveryDelay);
+            var errorContext = CreateErrorContext(numberOfDeliveryAttempts: 1);
+
+            var recoverabilityAction = policy.Invoke(errorContext);
+            var delayedRetryAction = recoverabilityAction as DelayedRetry;
+
+            Assert.IsNotNull(delayedRetryAction, "When immediate retries turned off and delayed retries left, recoverability policy should return DelayedRetry");
+            Assert.AreEqual(deliveryDelay, delayedRetryAction.Delay);
+        }
+
+        [Test]
+        public void When_immediate_retries_turned_off_and_slr_policy_returns_no_delay_should_return_move_to_errors_action()
```

Comment:
`When_max_immediate_retries_exceeded_should_return_action_other_than_immediate_retry` is incomplete. Renamed to `When_immediate_retries_turned_off_and_slr_policy_returns_delay_should_return_delayed_retry`. Added two more tests. In this case I think having the explicit tests is more expressive than test cases. Happy to be convinced otherwise though